src/server/handlers/mod.rs

```diff
@@ -18,19 +24,60 @@ use serde_json::Value;
 use std::collections::HashMap;
 use std::sync::Arc;
 
-pub type Method = String;
+/// A util macro to create the handlers and do the method routing.
+/// This macro generates the `handle` method. In the method, it just matches the handler with
+/// different enum types to get the inner handler instance. Every time a new handler is added,
+/// a new match statement must be copy-pasted. With this macro, we can reduce a lot of repetitive
+/// copy paste.
+macro_rules! route_handlers {
```

Comment:
Would you mind keeping this PR only for the implementation of the handlers and move the macro proposal to another PR? It may potentially require a longer discussion and I don't want to block these commands from being merged. Thanks!

Reply:
sure sure